x86/plat.c

```diff
@@ -138,11 +143,13 @@ cpu_type_get(void)
 
 	family = CPU_FAMILY(eax);
 	model = CPU_MODEL(eax);
+	ext_family = CPU_EXT_FAMILY(eax);
 	ext_model = CPU_EXT_MODEL(eax);
 
-	if (family == 6) {
-		model = (ext_model << 4) + model;
+	family = family + ext_family;
```

Comment:
The BaseFamily field of CPUID 0x1 EAX is already maxed out at 0xf for recent AMD processors (Family 17h and later) but my intention was to make this part of the code as generic as possible to facilitate future inclusions of x86 processors from other vendors.

Reply:
yeah, I see your purpose is to make this valid for all vendors, and it's ok for current processors e.g intel doesn't use ext_family so far I think. My point is just to ask if we should stick with spec definition more closely to just add ext_family in case of family=0xf.

Reply:
Got it.